### PR TITLE
samples: sensor: bme280: add README.rst

### PR DESCRIPTION
**Pull request by overheat**
Added BME280 (humidity and pressure sensor) sample documentation for
guiding beginners.

Signed-off-by: Aaron Tsui <aaron.tsui@outlook.com>

### CODE REVIEW

File: samples/sensor/bme280/README.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+`bme280 datasheet`_ at page 38.
+There are two processor cores (x86 and ARC) on the Arduino 101. You will need to
+flash both this sample's code on the ARC core (using the ``arduino101_ss`` board target),
+and stub code on the x86 core (using the ``arduino_101`` board target), as shown below.
+
+.. code-block:: console
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/sensors/bme280
+   :board: arduino_101_sss
```
Comment (MaureenHelm) [inline samples/sensor/bme280/README.rst]: This sample doesn't build out of the box for this board. Will you please also add a dts overlay to fix this?
```
In file included from /home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:24:
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.h:111:2: error: #error "BME280 device type not specified"
 #error "BME280 device type not specified"
  ^~~~~
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c: In function 'bme280_init':
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:357:22: warning: unused variable 'data' [-Wunused-variable]
  struct bme280_data *data = dev->driver_data;
                      ^~~~
In file included from ../../../../include/sensor.h:23,
                 from /home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:11:
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c: At top level:
/home/maureen/zephyrproject/zephyr/drivers/sensor/bme280/bme280.c:385:29: error: 'DT_BOSCH_BME280_0_LABEL' undeclared here (not in a function); did you mean 'DT_BOSCH_BMI160_0_LABEL'?
 DEVICE_AND_API_INIT(bme280, DT_BOSCH_BME280_0_LABEL, bme280_init, &bme280_data,
                             ^~~~~~~~~~~~~~~~~~~~~~~
../../../../include/device.h:107:11: note: in definition of macro 'DEVICE_AND_API_INIT'
   .name = drv_name, .init = (init_fn),     \
           ^~~~~~~~
```

File: samples/sensor/bme280/README.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        $ sudo picocom -D /dev/ttyUSB0
+
+.. code-block:: console
+
+        temp: 27.950000; press: 100.571027; humidity: 61.014648
+        temp: 27.940000; press: 100.570269; humidity: 61.012695
+        temp: 27.950000; press: 100.570695; humidity: 61.002929
+
+.. _bme280 datasheet: https://www.mouser.com/ds/2/621/BST-BME280_DS001-09-521021.pdf
```
Comment (MaureenHelm) [inline samples/sensor/bme280/README.rst]: Please link directly to the vendor: https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BME280-DS002.pdf

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks @overheat for adding this!

File: samples/sensor/bme280/README.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+.. zephyr-app-commands::
+   :zephyr-app: tests/booting/stub
+   :board: arduino_101
+   :goals: flash
+   :compact:
+
+Sample Output
+=============
+To check output of this sample , any serial console program can be used.
```
Comment (dbkinder) [inline samples/sensor/bme280/README.rst]:
```suggestion
To check the output of this sample, any serial console program can be used.
```

File: samples/sensor/bme280/README.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Here I am using picocom program to open output. Check which tty device it is.
```
Comment (dbkinder) [inline samples/sensor/bme280/README.rst]: Change to:
```
This example uses ``picocom`` on the serial port ``/dev/ttyUSB0``:
```

File: samples/sensor/bme280/README.rst [hunk 5]
```diff
@@ -0,0 +1,54 @@
+.. _bme280:
+
+BME280 Humidity and Pressure Sensor
+###################################
+
+Overview
+********
+
+This sample application periodically reads temperature, pressure and humidity data from
```
Comment (dbkinder) [inline samples/sensor/bme280/README.rst]:
```suggestion
This sample application periodically reads temperature, pressure, and humidity data from
```

File: samples/sensor/bme280/README.rst [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+the first available device that implements SENSOR_CHAN_AMBIENT_TEMP, SENSOR_CHAN_PRESS
```
Comment (dbkinder) [inline samples/sensor/bme280/README.rst]:
```suggestion
the first available device that implements SENSOR_CHAN_AMBIENT_TEMP, SENSOR_CHAN_PRESS,
```

File: samples/sensor/bme280/README.rst [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+and SENSOR_CHAN_HUMIDITY. This sample checks the sensor in polling mode (without
+interrupt trigger).
+
+Building and Running
+********************
+
+This sample application uses an BME280 sensor connected to an Arduino 101 board via I2C.
+Sensor has multiple pins so you need to connect according to connection diagram given in
```
Comment (dbkinder) [inline samples/sensor/bme280/README.rst]:
```suggestion
Connect the sensor pins according to the connection diagram given in the
```

File: samples/sensor/bme280/README.rst [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+`bme280 datasheet`_ at page 38.
+There are two processor cores (x86 and ARC) on the Arduino 101. You will need to
+flash both this sample's code on the ARC core (using the ``arduino101_ss`` board target),
+and stub code on the x86 core (using the ``arduino_101`` board target), as shown below.
+
+.. code-block:: console
```
Comment (dbkinder) [inline samples/sensor/bme280/README.rst]: delete this line (with the code-block directive) since there is no code block here.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/sensor/bme280/README.rst (hunk 1 above)
Comment (overheat) [reply samples/sensor/bme280/README.rst]: Ok, which board do your prefer? Arduino_101?

File: samples/sensor/bme280/README.rst (hunk 4 above)
Comment (overheat) [reply samples/sensor/bme280/README.rst]: @dbkinder Thank you for those suggestions. Should I need to change this file locally in my PC, and check it manually? Is there any way that accepts your commits and pull it down to my PC?

Comment (dbkinder) [reply samples/sensor/bme280/README.rst]: It seems to work better to make the change locally and do a "git commit --amend" and a forced push to update your PR.

File: samples/sensor/bme280/README.rst (hunk 1 above)
Comment (overheat) [reply samples/sensor/bme280/README.rst]: Done. Please help me to review again. Thanks.

**Review by MaureenHelm — APPROVED** [review]

File: samples/sensor/bme280/README.rst (hunk 4 above)
Comment (overheat) [reply samples/sensor/bme280/README.rst]: Done. Please help me to review again. Thanks.

Comment (overheat) [reply samples/sensor/bme280/README.rst]: @dbkinder

Comment (dbkinder) [reply samples/sensor/bme280/README.rst]: It doesn't appear that my previous comments were addressed yet?

**Review by dbkinder — CHANGES_REQUESTED** [review]
It doesn't appear my previous comments were addressed and  the PR updated?

**Comment by overheat** [comment]
@dbkinder Sorry for clicked the wrong button and let your comments became `Outdated` status. 
I think I forgot "git add" the file in last push action, so nothing updated for your comments were addressed.
I have modified and push again.

**Review by dbkinder — APPROVED** [review]
Some typo edits are still missing, but close enough.

File: samples/sensor/bme280/README.rst (hunk 2 above)
Comment (MaureenHelm) [reply samples/sensor/bme280/README.rst]: I thought you fixed this earlier, but it seems to have been reverted. The dts overlay disappeared too.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: samples/sensor/bme280/README.rst (hunk 2 above)
Comment (overheat) [reply samples/sensor/bme280/README.rst]: I am very sorry about this. I try to works on different PC but looks it not a good idea for me now. I have correct those issues.

**Comment by overheat** [comment]
@MaureenHelm Already modified. Please help me to review, thanks!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**